File: gfg/2024_6_june/python/08_index_of_an_extra_element.py
# CODE

class Solution:
    def findExtra(self, n, a, b):
        left = 0
        right = n - 1
        while left < right:
            mid = left + ((right - left) >> 1)
            if b[mid] == a[mid]:
                left = mid + 1
            else:
                right = mid
        return left

File: gfg/2024_6_june/c++/08_index_of_an_extra_element.cpp
// APPROACH
// 1 simple binary search question.
// 2 if element at mid of arr1 is found at mid of arr2 then start is mid + 1.
// 3 if element at mid of arr1 is not found at mid of arr2 then e = mid.
// 4 update mid.
// 5 return start.

// TIME COMPLEXITY
// O(logn)

// SPACE COMPLEXITY
// O(1)

// CODE

class Solution {
  public:
    int findExtra(int n, int arr1[], int arr2[]) {
        int st = 0; 
        int e = n - 1;
        int mid = st + (e - st) / 2;
        
        while(st < e){
            if(arr1[mid] == arr2[mid]){
                st = mid + 1;
            }
            else{
                e = mid;
            }
            mid = st + (e - st) / 2;
        }
        return st;
    }
};